### PR TITLE
Rexport types for servers

### PR DESCRIPTION
**Pull request by dvdplm**
Include the "macros" feature in the the "server" feature so users don't have to import "jsonrpsee-types" explicitly.

Also re-export the `SubscriptionSink`.

This is possibly slightly contentious. I think it's fair to assume that people writing servers will want to use the "macros" feature (which implies "types"). The downside of this is that "macros" includes the client side macros and "types" also includes types used in clients more.

Alternative solutions would be
- include the "types" feature in "server" instead (downside: you get both [client](https://github.com/paritytech/jsonrpsee/blob/dd4a17ba5e92091458805ad8017c2926cd28d1fa/types/src/client.rs) and server types)
- granularly re-export common types like `CallError` or `SubscriptionSink` (downside: more cfg-fu)
- `<insert-yours />`

Fixes https://github.com/paritytech/jsonrpsee/issues/407

Substrate "companion": https://github.com/paritytech/substrate/compare/dp-jsonrpsee-integration-2...dp-tweak-reexports?expand=1

### CODE REVIEW

File: jsonrpsee/Cargo.toml [hunk 1]
```diff
@@ -20,6 +20,6 @@ types = { path = "../types", version = "0.2.0", package = "jsonrpsee-types", opt
 
 [features]
 client = ["http-client", "ws-client"]
-server = ["http-server", "ws-server", "utils"]
+server = ["http-server", "ws-server", "utils", "macros"]
```
Comment (niklasad1) [inline jsonrpsee/Cargo.toml:23]: I think types should be sufficient here, no?

File: jsonrpsee/Cargo.toml [hunk 2]
```diff
@@ -20,6 +20,6 @@ types = { path = "../types", version = "0.2.0", package = "jsonrpsee-types", opt
 
 [features]
 client = ["http-client", "ws-client"]
```
Comment (niklasad1) [inline jsonrpsee/Cargo.toml:22]: perhaps we should add types here too? however types are re-exported from the clients

**Review by niklasad1 — COMMENTED** [review]
I think we should re-export the types from the servers too inorder for the crates to be used as independent crates but the most common usage should be from the facade crate anyway.

File: jsonrpsee/Cargo.toml (hunk 1 above)
Comment (dvdplm) [reply jsonrpsee/Cargo.toml:23]: "types" is sufficient, but I'm a bit concerned that [this](https://github.com/paritytech/jsonrpsee/blob/dd4a17ba5e92091458805ad8017c2926cd28d1fa/jsonrpsee/src/lib.rs#L28-L35) will become messy. I'd like to be able to look at `lib.rs` and see at a glance where everything is coming from.

File: http-server/src/lib.rs [hunk 3]
```diff
@@ -40,7 +40,7 @@ pub use access_control::{
 	hosts::{AllowHosts, Host},
 	AccessControl, AccessControlBuilder,
 };
-pub use jsonrpsee_types::{Error, TEN_MB_SIZE_BYTES};
+pub use jsonrpsee_types::*;
```
Comment (maciejhirsz) [inline http-server/src/lib.rs:43]: Wouldn't it make more sense to re-export types in namespace like: `pub use jsonrpsee_types as types`?

Comment (dvdplm) [reply http-server/src/lib.rs]: Yes I agree. I went ahead and did that so it's a lot more noise here now.

**Review by maciejhirsz — APPROVED** [review]

File: jsonrpsee/src/lib.rs [hunk 4]
```diff
@@ -18,14 +18,14 @@ pub use ws_server;
...
-pub use utils::server::rpc_module::RpcModule;
+pub use utils::server::rpc_module::{RpcModule, SubscriptionSink};
 
 /// Procedural macros for JSON RPC implementations.
 #[cfg(feature = "macros")]
 pub use proc_macros;
 
 /// Common types used to implement JSON RPC server and client.
-#[cfg(feature = "macros")]
+#[cfg(any(feature = "types", feature = "macros"))]
```
Comment (niklasad1) [inline jsonrpsee/src/lib.rs:28]: :+1:

File: ws-client/src/lib.rs [hunk 5]
```diff
@@ -27,4 +27,4 @@ mod tokio;
 mod tests;
 
 pub use client::{WsClient, WsClientBuilder};
-pub use jsonrpsee_types::*;
+pub use jsonrpsee_types as types;
```
Comment (niklasad1) [inline ws-client/src/lib.rs:30]: :+1

**Review by niklasad1 — APPROVED** [review]

**Pull request merged**